rust-toolchain.toml

```diff
@@ -0,0 +1,5 @@
+# Pinning our toolchain ensures ./scripts/regenerate_test_rustdocs.sh always
+# emits the desired format.
+
+[toolchain]
+channel = "nightly-2023-08-18"
```

Comment:
```suggestion
channel = "1.73"
```
This corresponds to stable Rust, so let's use the newest stable Rust that matches — it'll be mentioned in CI.

I wonder if this might cause a problem when CI tries to test this with multiple Rust versions 🤔

Reply:
Filed https://github.com/actions-rust-lang/setup-rust-toolchain/issues/28 since CI ignores our explicit toolchain command and uses what's in the file — in the meantime, to unblock this PR, let's add a step to CI to delete the `rust-toolchain.toml` file before setting up Rust with that action.

Reply:
Yeah . . . should I hold off on PR'ing this?

Reply:
Maybe put it on the back burner for a ~day or two. It seems like the GitHub action might get an update that would fix this behavior.

It might be good to open a draft PR for rustdoc-v27, just to make sure we don't forget to update it there. Then, assuming the action gets an update, I can update it in all three branches and both this PR and the rustdoc-v27 one would become directly mergeable.

Reply:
Sounds good.

Reply:
The action got updated, so I'm going to re-trigger the tests here and I expect they should pass.

Reply:
Hmm tests failed on output mismatch errors 🤔 Not sure what's going on.

Since it's your PR, I'll let you look into it unless you specifically want me to take over.

Reply:
I'll look into it tomorrow

Reply:
Resolving as stale. (new comments in discussion below)